OpenRA.Mods.D2k/Traits/World/BuildableTerrainLayer.cs

```diff
@@ -15,22 +15,35 @@
 namespace OpenRA.Mods.D2k.Traits
 {
 	[Desc("Attach this to the world actor. Required for LaysTerrain to work.")]
-	public class BuildableTerrainLayerInfo : TraitInfo<BuildableTerrainLayer> { }
-	public class BuildableTerrainLayer : IRenderOverlay, IWorldLoaded, ITickRender
+	public class BuildableTerrainLayerInfo : ITraitInfo
 	{
-		Dictionary<CPos, Sprite> tiles;
-		Dictionary<CPos, Sprite> dirty;
+		[Desc("Palette to render the layer sprites in.")]
+		public readonly string Palette = "terrain";
+
+		public object Create(ActorInitializer init) { return new BuildableTerrainLayer(init.Self, this); }
+	}
+
+	public class BuildableTerrainLayer : IRenderOverlay, IWorldLoaded, ITickRender, INotifyActorDisposing
+	{
+		readonly BuildableTerrainLayerInfo info;
+		readonly Dictionary<CPos, Sprite> dirty = new Dictionary<CPos, Sprite>();
+		readonly TileSet tileset;
+		readonly Map map;
+
+		TerrainSpriteLayer render;
 		Theater theater;
-		TileSet tileset;
-		Map map;
+
+		public BuildableTerrainLayer(Actor self, BuildableTerrainLayerInfo info)
+		{
+			this.info = info;
+			tileset = self.World.TileSet;
+			map = self.World.Map;
+		}
 
 		public void WorldLoaded(World w, WorldRenderer wr)
 		{
 			theater = wr.Theater;
-			tileset = w.TileSet;
-			map = w.Map;
-			tiles = new Dictionary<CPos, Sprite>();
-			dirty = new Dictionary<CPos, Sprite>();
+			render = new TerrainSpriteLayer(w, wr, theater.Sheet, BlendMode.Alpha, wr.Palette(info.Palette), wr.World.Type != WorldType.Editor);
```

Comment:
Technically, sure, but it doesn't make a lot of sense.  This trait renders normal terrain tiles which by definition use the alpha blend mode.